src/main/java/de/sakpaas/backend/service/OccupancyService.java

```diff
@@ -172,4 +185,59 @@ public void checkReportLimit(Location location, byte[] requestHash)
   public Occupancy save(Occupancy occupancy) {
     return occupancyRepository.save(occupancy);
   }
+
+
+  /**
+   * Gets the Occupancy from recently submitted reports. Is Null if there are no in the range.
+   *
+   * @param location The Location
+   * @return The Occupancy
+   */
+  private AccumulatedOccupancy getLiveOccupancy(Location location) {
+    ZonedDateTime time = now();
+    List<Occupancy> occupancies = occupancyRepository.findByLocationAndTimestampAfter(location,
+        now().minusMinutes(config.getDuration()));
+
+    return new AccumulatedOccupancy(
+        calculateAccumulatedOccupancy(occupancies, time),
+        occupancies.size(),
+        occupancies.stream()
+            .map(Occupancy::getTimestamp)
+            .max(Comparator.naturalOrder())
+            .orElse(null)
+    );
+  }
+
+  /**
+   * Returns the hour of the week, based on the Date. Public Holidays are treated like a sunday.
+   *
+   * @param date The Date
+   * @return The hour of the week
```

Comment:
Was soll die `hour of the week` sein?

Reply:
In welcher Stunde der Woche sich das Datum+Uhrzeit befindet.

Beispielsweise wäre Montag 13:30 die Wochenstunde 13, Dienstag 13:30 wäre hingegen 27. In diesem Beispiel würde es sich nach Sonntag wieder zurück setzten, also natürlich abhängig welches der erste Tag der Woche ist.